core/domain/user_jobs_one_off.py

```diff
@@ -130,3 +134,93 @@ def reduce(key, stringified_values):
                 subscription_services.subscribe_to_exploration(key, item['id'])
             elif item['type'] == 'collection':
                 subscription_services.subscribe_to_collection(key, item['id'])
+
+
+class UserImpactCalculationOneOffJob(jobs.BaseMapReduceJobManager):
+    """ Calculates each user's impact score, where impact score is defined as:
+    Sum of (
+    ln(playthroughs) * (ratings_scaler) * (average(ratings) - 2.5))
+    *(multiplier),
+    where multiplier = 10, and ratings_scaler is .1(number of ratings)
+    if there are < 10 ratings for that exploration.
+
+    Impact scores are calculated over explorations for which a user
+    is listed as a contributor.
+    """
+
+    MULTIPLIER = 10
+    MIN_AVERAGE_RATING = 2.5
+    NUM_RATINGS_SCALER_CUTOFF = 10
+    NUM_RATINGS_SCALER = .1
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @classmethod
+    def _get_exp_impact_score(cls, exploration_id):
+        from core.domain import rating_services
+        from core.domain import stats_jobs_continuous
+
+        # Get ratings and compute average rating score.
+        ratingFrequencies = rating_services.get_overall_ratings_for_exploration(
```

Comment:
I think I just copied this pattern from another file-- I'll change it into underscores!